nameable.rb

```diff
@@ -0,0 +1,9 @@
+class Nameable
+  def initialize(nameable)
+    @nameable = nameable
+  end
+
+  def correct_name
+    raise NotImplementedError
```

Comment:
- **[OPTIONAL]** It would be nice to add a message that provides more detail for where this error is raised.

  ```
  raise NotImplementedError, "#{self.class} has not implemented method '#{__method__}'"
  ```